### PR TITLE
#162170720 Add pagination to project

### PR DESCRIPTION
**Pull request by okoroemeka**
#### What does this PR do?
Have pagination support for Author-haven

#### Description of Task to be completed?
Have pagination working on the following endpoint
`GET api/V1/articles`

#### How should this be manually tested?
After cloning the repo, cd into it, run `npm install` then run `npm dev:start`

* Use Postman to test the route

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#162170720](https://www.pivotaltracker.com/story/show/162170720)

#### Screenshots (if appropriate)
screen shot 2018-12-11 at 11 42 48 pm

#### Questions:
N/A

### CODE REVIEW

File: src/helpers/paginationHelper.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+   ** @description - This method is responsible for verifying if a query parameter is negative
+   * @param {*} request
+   * @param {*} response
+   * @param {*} next
+   * @returns {object} - object representing response message
+   */
+  static checkQueryparameter(request, response, next) {
+    const { page } = request.query;
+    if (page < 1) {
+      return response.status(400).json({
```
Comment (frostyblok) [inline src/helpers/paginationHelper.js]: Hello Emeka, I like your implementation and how readable your code is but I have a feedback. There is a helper function in the helper folder that would help display error messages properly. You can find that [here](https://github.com/andela/balder-ah-backend/blob/staging/src/helpers/index.js).

File: test/unit/article/controllers/articleController.spec.js [hunk 2]
```diff
@@ -84,21 +84,6 @@ describe('Article Controller', () => {
   });
 
   describe('getAllArticles()', () => {
```
Comment (frostyblok) [inline test/unit/article/controllers/articleController.spec.js:86]: Also, Emeka, why did you remove the test case that was here initially?

File: src/helpers/paginationHelper.js (hunk 1 above)
Comment (okoroemeka) [reply src/helpers/paginationHelper.js]: Resolved.

File: test/unit/article/controllers/articleController.spec.js (hunk 2 above)
Comment (okoroemeka) [reply test/unit/article/controllers/articleController.spec.js:86]: The test was failing and I couldn't figure out why. I raised that PR as `WIP`. But the test has been fixed and the problem resolved.

Comment (frostyblok) [reply test/unit/article/controllers/articleController.spec.js:86]: OK, good job with working with the feedback!

**Review by frostyblok — APPROVED** [review]

File: src/controllers/article.js [hunk 3]
```diff
@@ -50,8 +50,9 @@ class ArticleController {
    */
   static async getAllArticles(request, response) {
     try {
-      const allArticles = await ArticleModel.getAllArticle();
-      if (!allArticles) {
+      const { page } = request.query;
+      const allArticles = await ArticleModel.getAllArticle(page);
+      if (!allArticles || !allArticles.length) {
```
Comment (akhilome) [inline src/controllers/article.js]: `!allArticles || !allArticles.length` can be just `!allArticles.length`, don't you think?

File: src/helpers/articles.js [hunk 4]
```diff
@@ -21,13 +23,25 @@ class ArticleModel {
   /**
    * @description - This method is responsible for querying all articles from the database
    * @static
-   * @param {object} request - Request sent to the router
+   * @param {object} page - Request sent to the router
```
Comment (akhilome) [inline src/helpers/articles.js]: `page - Request sent to the router` <= Don't you think this parameter's description is not correct?

File: src/helpers/articles.js [hunk 5]
```diff
@@ -21,13 +23,25 @@ class ArticleModel {
...
    * @param {object} response - Response sent from the controller
```
Comment (akhilome) [inline src/helpers/articles.js]: This method doesn't accept a `response` parameter. This should be taken out.

File: src/helpers/paginationHelper.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ */
+class paginationHelper {
+  /**
+   * @description - This method is responsible for returning numbeer of article per page
+   * @static
+   * @param {number} number - parameter sent to controller for get all article
+   * @returns {number} - object representing response message
+   * @memberof paginationHelper
+   */
+  static numberOfArticlePerPage(number) {
```
Comment (luqmanoop) [inline src/helpers/paginationHelper.js]: Nice work @okoroemeka, looking at this function, it doesn't seem to have any logic inside it. I think it would be better to remove it.

**Review by luqmanoop — CHANGES_REQUESTED** [review]
Hi @okoroemeka, I left a comment [here](https://github.com/andela/balder-ah-backend/pull/34/files#r240947686). Please attend to it.

File: src/helpers/paginationHelper.js (hunk 6 above)
Comment (akhilome) [inline src/helpers/paginationHelper.js]: How come you're receiving the `number` parameter for the method, just to return it. Please, can you throw more light on what you're trying to achieve with this method?

File: src/helpers/paginationHelper.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  /**
+   ** @description - This method is responsible for verifying if a query parameter is negative
+   * @param {*} request
+   * @param {*} response
+   * @param {*} next
+   * @returns {object} - object representing response message
+   */
+  static checkQueryparameter(request, response, next) {
```
Comment (akhilome) [inline src/helpers/paginationHelper.js:15]: Good forward thinking with this. I, however, think you can take this further by ensuring "strings" don't go through, too.

**Review by akhilome — CHANGES_REQUESTED** [review]
@okoroemeka, nice job delivering this.

I love the fact that you've taken the feedback @waleCloud gave you on "commit message" descriptions to heart.

I just left a few comments on your code, please go through them.

File: src/helpers/paginationHelper.js (hunk 6 above)
Comment (okoroemeka) [reply src/helpers/paginationHelper.js]: That's true thanks for pointing that out. I have removed it.

Comment (okoroemeka) [reply src/helpers/paginationHelper.js]: Thanks for pointing that out. However, I have removed it.

File: src/helpers/paginationHelper.js (hunk 7 above)
Comment (okoroemeka) [reply src/helpers/paginationHelper.js:15]: Thanks for pointing out, it didn't cross my mind to do that. I will Implement it now.

File: src/helpers/articles.js (hunk 5 above)
Comment (okoroemeka) [reply src/helpers/articles.js]: Thanks. I will take care of it.

File: src/helpers/articles.js (hunk 4 above)
Comment (okoroemeka) [reply src/helpers/articles.js]: Thanks for pointing this out also.

File: src/helpers/paginationHelper.js (hunk 7 above)
Comment (okoroemeka) [reply src/helpers/paginationHelper.js:15]: Resolved.

File: src/helpers/articles.js (hunk 5 above)
Comment (okoroemeka) [reply src/helpers/articles.js]: Resolved.

File: src/helpers/articles.js (hunk 4 above)
Comment (okoroemeka) [reply src/helpers/articles.js]: Resolved.

File: src/controllers/article.js (hunk 3 above)
Comment (okoroemeka) [reply src/controllers/article.js]: I just learnt a new thing from this. Thanks for pointing it out.

**Review by luqmanoop — APPROVED** [review]

**Review by akhilome — APPROVED** [review]
